ArduPlane/ArduPlane.cpp

```diff
@@ -963,6 +963,28 @@ bool Plane::flight_option_enabled(FlightOptions flight_option) const
     return g2.flight_options & flight_option;
 }
 
+#if AP_SCRIPTING_ENABLED
+// this implements the virtual function defined in AP_Vehicle for Plane to set the desired airspeed in m/s
+bool Plane::set_desired_airspeed(float airspeed_new)
+{
+    if (control_mode->is_guided_mode() || control_mode == &mode_auto) {
+        plane.new_airspeed_cm = constrain_float(airspeed_new, aparm.airspeed_min, aparm.airspeed_max) * 100.0f;
+        return true;
+    }
+    return false;
+}
+
+// Helper function to let scripting set the guided mode loiter radius and direction
+bool Plane::set_guided_radius_and_direction(float radius, bool direction_is_ccw)
```

Comment:
I would prefer to have bindings to the functionality under the OFFBOARD_GUIDED code in plane, which is currently accessed via MAVLink with MAV_CMD_GUIDED_xxx funtions.
That will give a lot more functionality and re-uses existing tested capability.
The set_guided_radius_and_direction() method is really just a workaround for not having the OFFBOARD_GUIDED APIs

Reply:
Ive removed this. I'll use the MAVLink API.